components/polymer/dust-demo.html

```diff
@@ -127,23 +102,33 @@ <h3 id="output-header" on-click="{{collapseContainer}}">Output</h3>
         this.renderDust();
       },
       renderDust: function() {
-        var templateEl = this.querySelector('dust-demo-template');
-        var jsonEl = this.querySelector('dust-demo-json');
-        if (!(templateEl && jsonEl)) {
+        var templateEditorEl, jsonEl,
+            template, data,
+            compiledContainer, outputContainer,
+            compiledDust;
+
+        if (this.$.template.getDistributedNodes()) {
+          templateEditorEl = this.$.template.getDistributedNodes()[0].$.workspace;
+        }
+        if (!templateEditorEl) {
           return;
         }
         // Unescape '>' and '<' to read Dust's include syntax correctly
-        var template = templateEl.innerHTML.replace(/&gt;/g, '>').replace(/&lt;/g, '<');
+        var template = templateEditorEl.value.replace(/&gt;/g, '>').replace(/&lt;/g, '<');
         if (!template) {
           return;
         }
-        var data = jsonEl.innerHTML.replace('&nbsp;', ' ');
-        if (data) {
-          eval('var data = ' + data + ';');
+
+        // JSON container is optional
+        jsonEl = this.querySelector('dust-demo-json');
+        if (jsonEl) {
+          data = jsonEl.innerHTML.replace('&nbsp;', ' ');
         }
-        var compiledContainer = this.$['compiled-dust'];
-        var outputContainer = this.$.output;
-        var compiledDust = dust.compile(template, this.templateName);
+        data = data ? JSON.parse(data) : {};
```

Comment:
I think this will fail when you have functions in your data (and functions are valid). I think the ugly `eval` is necessary... :(